validates.js

```diff
@@ -1,67 +1,93 @@
 import React from 'react';
+import { string, func, element, oneOf } from 'prop-types';
+
+const undef = void 0;
+function noop() {}
 
 export default class Validates extends React.Component {
+  get propsHandler() {
+    const { onValidChange = noop } = this.props;
+    return onValidChange;
+  }
+
+  get ctxHandler() {
+    const { onValidChange = noop } = this.context;
+    return onValidChange;
+  }
```

Comment:
You only use these getters once in `onValidChange`. Remove them and simply define them in that function scope.